rfcs/react-components/styles-handbook.md

```diff
@@ -0,0 +1,922 @@
+
+```js
+import { makeStyles, shorthands } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    ...shorthands.padding('4px'),
+    ...shorthands.margin('4px'),
+
+    ':hover': {
+      ...shorthands.padding('4px'),
+      ...shorthands.margin('4px'),
+    },
+
+    '::before': {
+      display: 'block',
+      content: "' '",
+    },
+
+    '@media (forced-colors: active)': {
+      ...shorthands.padding('4px'),
+      ...shorthands.margin('4px'),
+    },
+  },
+});
+// ⬇️⬇️⬇️
+// produces 26 classes ((4+4)+(4+4)+2+(4+4))
+```
+
+Such cases might be unavoidable by design, [the next section](#makeresetstyles) covers APIs to address this problem.
+
+## `mergeClasses()`
+
+The [`mergeClasses() API`][griffel-merge-classes] should be used when multiple Griffel styles are used on the same element.
+
+```js
+import { makeStyles, shorthands } from '@griffel/react';
+
+const useClasses = makeStyles({
+  root: {
+    /* styles */
+  },
+  foo: {
+    /* styles */
+  },
+  bar: {
+    /* styles */
+  },
+});
+
+function Component(props) {
+  const classes = useClasses();
+  const className = mergeClasses(
+    classes.root,
+    props.foo && classes.foo /* styles specific for "foo" */,
+    props.bar && classes.bar /* styles specific for "bar" */,
+  );
+
+  /* --- */
+}
+```
+
+### Order of arguments determines results
+
+Unlike native CSS, the output of mergeClasses() is affected by the order of the classes passed in, allowing for control over priority of style overrides.
+
+```js
+import { mergeClasses, makeStyles } from '@griffel/react';
+
+const useClasses = makeStyles({
+  blue: { color: 'blue' },
+  red: { color: 'red' },
+});
+
+function Component(props) {
+  // ℹ️ Order of arguments determines the results
+
+  const redClassName = mergeClasses(classes.blue, classes.red);
+  // 👆 { color: 'red' }
+  const blueClassName = mergeClasses(classes.red, classes.blue);
+  // 👆 { color: 'blue' }
+
+  /* --- */
+}
+```
+
+### ⚠️ Only combine classes with `mergeClasses`
+
+It is not possible to simply concatenate classes returned by `useClasses()` hooks. Always use `mergeClasses()` to merge classes as results of concatenation can contain duplicated classes and lead to non-deterministic results.
+
+```js
+import { makeStyles, mergeClasses } from '@griffel/react';
+
+const useClasses = makeStyles({
+  rootA: { display: 'flex' },
+  rootB: { display: 'grid' },
+});
+
+function App(props) {
+  const classes = useClasses();
+
+  // ✅ Returns "class-display-grid"
+  const correctClasses = mergeClasses(classes.rootA, classes.rootB);
+  // 🔴 Never concatenate class strings, returns "class-display-flex class-display-grid"
+  const wrongClasses = classes.rootA + ' ' + classes.rootB;
+}
+```
+
+## `makeResetStyles`
+
+This API works similarly to `makeStyles` and is used to styles as a single monolithic class to avoid the "CSS rules explosion" problem.
```

Comment:
```suggestion
This API works similarly to `makeStyles` and is used to generate styles as a single monolithic class to avoid the "CSS rules explosion" problem.
```